File: webapp/services/user/internal/http/jsend.go
//
// user/internal/http/jsend.go
//

package http

type JSend struct {
	Status  string      `json:"status"`
	Data    interface{} `json:"data,omitempty"`
	Message string      `json:"message,omitempty"`
}

func Success(data interface{}) JSend {
	return JSend{
		Status: "success",
		Data:   data,
	}
}

func Fail(msg string) JSend {
	return JSend{
		Status:  "fail",
		Message: msg,
	}
}

func Error(msg string) JSend {
	return JSend{
		Status:  "error",
		Message: msg,
	}
}

File: webapp/services/auth/app/jsend.py
#
# auth/app/jsend.py
#

from fastapi.responses import JSONResponse


def success(data=None, status_code=200):
    payload = {"status": "success"}

    if data is not None:
        payload["data"] = data

    return JSONResponse(status_code=status_code, content=payload)


def fail(message, status_code=400):
    return JSONResponse(
        status_code=status_code,
        content={"status": "fail", "message": message},
    )


def error(message="Internal server error", status_code=500):
    return JSONResponse(
        status_code=status_code,
        content={"status": "error", "message": message},
    )
